### PR TITLE
#668 - Username and password support in Maven proxy

### PR DESCRIPTION
**Pull request by olegmoz**
Part of #668 
Username and password support in Maven proxy

### CODE REVIEW

File: src/main/java/com/artipie/maven/MavenProxy.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            if (password == null) {
+                throw new IllegalStateException(
+                    "`password` is not specified in settings for Maven proxy"
+                );
+            }
+            auth = new GenericAuthenticator(username, password);
+        }
+        return new MavenProxySlice(
+            this.client,
+            URI.create(settings.string("remote_uri")),
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/maven/MavenProxy.java:96]: @olegmoz I'd suggest to check whether `remote_uri` is present and throw corresponding exception if not

**Review by olenagerasimova — CHANGES_REQUESTED** [review]
@olegmoz just one suggestion, take a look please

**Comment by olegmoz** [comment]
@olenagerasimova thanks, corrected

**Review by olenagerasimova — APPROVED** [review]
@olegmoz thanks

**Comment by 0crat** [comment]
Job `gh:artipie/artipie#672` is not assigned, can't get performer

**Pull request merged**